### ISSUE TITLE
progress bar ignores log level

### ISSUE THREAD

**Issue by bsipocz** [astropy/astropy#7626]
We've received a report in astroquery that the progressbar doesn't take into account the log level and is always shown. (https://github.com/astropy/astroquery/issues/1179)

I think this is a very reasonable request and best to be address in astropy rather than downstream. The astroquery usage is quite standard: https://github.com/astropy/astroquery/blob/master/astroquery/query.py#L277

cc @texadactyl

**Comment by texadactyl** [comment]
The original discussion which inspired this issue report can be found here: https://github.com/KeplerGO/lightkurve/issues/165

It seems to me that the progress bar display ("downloading") could be viewed as equivalent to "Found cached file ..." from Astroquery (log.info); both result from the same high-level Astroquery query function, the difference being whether the file was or was not already cached.

On the other hand, I realize that there is always impact to any change that might be made.  So, I am sympathetic.

**Comment by pllim** [comment]
According to https://github.com/astropy/astropy/blob/master/astropy/utils/console.py#L982
```python
        file : writable file-like object, optional
            The file to write the to.  Defaults to `sys.stdout`.  If
            ``file`` is not a tty (as determined by calling its `isatty`
            member, if any), only ``msg`` will be displayed: the
            `ProgressBar` or `Spinner` will be silent.
```
In particular:
```python
color_print(msg, color, file=file)
```
So, indeed it was not designed to be smart about logger settings. Instead, you can pipe `file` to some `/dev/null` if you want to silence the message.

**Comment by texadactyl** [comment]
In this case, the lightkurve Python program isn't making the call to ProgressBar.  The intermediary, Astroquery query.py is actually making the call.  Are you suggesting that query interrogate the current effective log level?

By the way, I noticed in astropy.utils.data, that download_file & download_files_in_parallel (similar function to query) do have an option to suppress the ProgressBar.

@bsipocz - your opinion?

**Comment by pllim** [comment]
Since `astroquery` was calling the progress bar in a way that "resembles `astropy.utils.data.download_file` but uses the local `_session`," I think it would make sense for `astroquery` to implement the option to suppress progress bar over at `astroquery` side?

**Comment by texadactyl** [comment]
@pllim Agreed.
@bsipocz Maybe a new parameter on the query which defaults to calling ProgressBar E.g. show_progress=True ?  

I am not sure if the lightkurve folks are following this discussion.
@christinahedges or @barentsen - want to comment?

**Comment by bsipocz** [comment]
yeah, actually astroquery could indeed refactor the file download either to provide the alternative of using `requests` in astropy.utils, or switch to use `download_file`, but also the progressbar could be smarter out of the box for the greater good. 
But then we have https://github.com/astropy/astropy/issues/4738, so any efforts willing to be put into progress bar should probably be better to be used to implement the switch that issue suggest.

I'll reopen the astroquery issue then, so anyone is welcome to pick that up, too.

**Comment by pllim** [comment]
If astropy/astroquery#1188 is sufficient, we can close this here. Refactoring the entire progress bar is battle for another day...

**Comment by bsipocz** [comment]
@pllim - I would rather keep this issue open, with low-priority. I guess it's a good `package-novice` but not python novice issue.

(actually I think it's just `color_print` that needs to be log level aware).

**Comment by pllim** [comment]
But does it? `color_print` is a very low level operation. Having it depend on something high level like package-wide logger can come back to bite us. I think we need to think carefully about this first. I wouldn't advise marking this as novice.
